proposals/0295-block-limits-overage-relaxation.md

```diff
@@ -0,0 +1,81 @@
+---
+simd: '0295'
+title: Relax CU limit overage
+authors: Tao Zhu (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-06-04
+feature:
+---
+
+## Summary
+
+This proposal introduces a modification to block CU (Compute Unit) limit
+enforcement during block replay, allowing blocks that exceed the CU limit to be
+skipped without marking them as invalid and being rejected. This is an enabler
+for asynchronous block execution, where voting occurs prior to execution.
+
+## Motivation
+
+Current block execution behavior:
+
+- Transactions to be fully executed before votes are cast,
+- Each transaction's CU usage is tracked,
+- A block to be marked invalid during replay if the total consumed CU exceeds
+  the configured block CU limit.
+
+This behavior ensures strong guarantees around CU consumption but is not
+compatible with the upcoming async execution model, where replicas vote before
+execution.
+
+To support async execution, the system must gracefully handle the possibility
+that a valid voted-on block may exceed limits at replay time, without
+invalidating consensus.
+
+It proposes a relaxed approach for block CU enforcement: skip, instead of
+reject, a valid voted-on block that exceeds CU limit during replay.
+
+## New Terminology
+
+Skipped Block: a valid, committed block in terms of ledger continuity and votes,
+but all transactions in it are treated as having failed, therefore no state
+changes at all, nor fee collection, nor nonce advancement.
+
+## Detailed Design
+
+Change the block CU validation logic during replay as follows:
+
+- If a block exceeds the Block CU limit during replay execution:
+  - Stop execution of remaining transactions in the block, and rollback all
```

Comment:
Alpenglow has a similar construct where a block can become stateless midway (and later restarted off a different parent). I don't think this validation logic will interfere, just something to note.